aya/src/programs/mod.rs

```diff
@@ -805,6 +818,61 @@ impl_fd!(
     CgroupDevice,
 );
 
+/// Defines which [`Program`]s support the kernel's
+/// generic multi-prog API.
+///
+/// # Minimum kernel version
+///
+/// The minimum kernel version required to use this feature is 6.6.
+pub trait MprogProgram {
+    /// Returns a borrowed reference to the file descriptor of a given
+    /// [`Program`] which has support for the kernel's generic multi-prog API.
+    fn fd(&self) -> Result<BorrowedFd<'_>, ProgramError>;
+}
+
+macro_rules! impl_mprog_fd {
+    ($($struct_name:ident),+ $(,)?) => {
+        $(
+            impl MprogProgram for $struct_name {
+                /// Returns the a borrowed reference file descriptor of this Program.
+                fn fd(&self) -> Result<BorrowedFd<'_>, ProgramError> {
+                    Ok(self.fd()?.as_fd())
+                }
+            }
+        )+
+    }
+}
+
+impl_mprog_fd!(SchedClassifier,);
+
+/// Defines which [`Link`]s support the kernel's
+/// generic multi-prog API.
+///
+/// # Minimum kernel version
+///
+/// The minimum kernel version required to use this feature is 6.6.
+pub trait MprogLink {
```

Comment:
Should probably be called MultiProgLink? As a general rule, if the kernel uses
name X for something, use something that is as far as possible from X for the
aya API :P

Reply:
LOL fair enough, again I suck at naming :)